core/chains/evm/client/chain_client.go

```diff
@@ -294,53 +352,75 @@ func (c *chainClient) SendTransactionReturnCode(ctx context.Context, tx *types.T
 }
 
 func (c *chainClient) SequenceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (evmtypes.Nonce, error) {
-	return c.multiNode.SequenceAt(ctx, account, blockNumber)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return 0, err
+	}
+	return rpc.SequenceAt(ctx, account, blockNumber)
 }
 
 func (c *chainClient) SubscribeFilterLogs(ctx context.Context, q ethereum.FilterQuery, ch chan<- types.Log) (s ethereum.Subscription, err error) {
-	rpc, err := c.multiNode.SelectNodeRPC()
+	rpc, err := c.multiNode.SelectRPC()
 	if err != nil {
 		return s, err
 	}
 	return rpc.SubscribeFilterLogs(ctx, q, ch)
 }
 
-func (c *chainClient) SubscribeNewHead(ctx context.Context, ch chan<- *evmtypes.Head) (ethereum.Subscription, error) {
-	csf := newChainIDSubForwarder(c.ConfiguredChainID(), ch)
-	err := csf.start(c.multiNode.Subscribe(ctx, csf.srcCh, "newHeads"))
+func (c *chainClient) SubscribeNewHead(ctx context.Context) (<-chan *evmtypes.Head, ethereum.Subscription, error) {
+	rpc, err := c.multiNode.SelectRPC()
```

Comment:
Maybe not in scope of this PR, but we should check what would happen if this rpc gets outdated or is no longer a primary node after some time.
Does it mean the caller still is subscribed to an old rpc, while the MultiNode has switched to a newer one?

Do we know which clients use this function currently?

Reply:
Currently the client will keep the same RPC subscription and it's up to them to determine if they need to acquire a new subscription and ask MultiNode for which RPC to use next.  However, we can change this in the future to decentralize the subscriptions more by using multiple RPCs or perhaps expire them after some time.

Will resolve this for now and keep a note for future implementations.